src/components/uri/Visualization.tsx

```diff
@@ -204,14 +206,19 @@ export class VisualizationWrapped
             identifier,
             filters
         );
+
+        const featureFlags = await this.sdk.project.getFeatureFlags(this.props.projectId);
+        this.setState({ colorPaletteEnabled: Boolean(featureFlags.enableColorPalette) }, this.getColorPalette);
     }
 
     public componentWillUnmount() {
         this.subject.unsubscribe();
     }
 
     public shouldComponentUpdate(nextProps: IVisualizationProps, nextState: IVisualizationState) {
-        return this.hasChangedProps(nextProps) || (this.state.isLoading !== nextState.isLoading);
+        return this.hasChangedProps(nextProps)
+            || (this.state.isLoading !== nextState.isLoading)
+            || this.state.colorPalette !== nextState.colorPalette;
```

Comment:
colorPalette is array so checking by !== compares references. Is that what we want or should we use `_.isEqual`?